src/API/Endpoints/Migrations/RunMigration.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Give\API\Endpoints\Migrations;
+
+use Exception;
+use WP_REST_Request;
+use WP_REST_Response;
+use Give\MigrationLog\MigrationLogStatus;
+use Give\MigrationLog\MigrationLogFactory;
+use Give\Framework\Migrations\MigrationsRegister;
+
+/**
+ * Class RunMigration
+ * @package Give\API\Endpoints\Migrations
+ *
+ * @since 2.10.0
+ */
+class RunMigration extends Endpoint {
+
+	/** @var string */
+	protected $endpoint = 'migrations/run-migration';
+
+	/**
+	 * @var MigrationsRegister
+	 */
+	private $migrationRegister;
+
+	/**
+	 * @var MigrationLogFactory
+	 */
+	private $migrationLogFactory;
+
+	/**
+	 * RunMigration constructor.
+	 *
+	 * @param  MigrationsRegister  $migrationsRegister
+	 * @param  MigrationLogFactory  $migrationLogFactory
+	 */
+	public function __construct(
+		MigrationsRegister $migrationsRegister,
+		MigrationLogFactory $migrationLogFactory
+	) {
+		$this->migrationRegister   = $migrationsRegister;
+		$this->migrationLogFactory = $migrationLogFactory;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function registerRoute() {
+		register_rest_route(
+			'give-api/v2',
+			$this->endpoint,
+			[
+				[
+					'methods'             => 'POST',
+					'callback'            => [ $this, 'handleRequest' ],
+					'permission_callback' => [ $this, 'permissionsCheck' ],
+					'args'                => [
+						'id' => [
+							'validate_callback' => function( $param ) {
+								return ! empty( trim( $param ) );
+							},
+						],
+					],
+				],
+				'schema' => [ $this, 'getSchema' ],
+			]
+		);
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getSchema() {
+		return [
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'logs',
+			'type'       => 'object',
+			'properties' => [
+				'id' => [
+					'type'        => 'string',
+					'description' => esc_html__( 'Migration ID', 'give' ),
+				],
+			],
+		];
+	}
+
+	/**
+	 * @param  WP_REST_Request  $request
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function handleRequest( WP_REST_Request $request ) {
+		global $wpdb;
+		$migrationId  = $request->get_param( 'id' );
+		$migrationLog = $this->migrationLogFactory->make( $migrationId );
+
+		// Begin transaction
+		$wpdb->query( 'START TRANSACTION' );
+
+		try {
+			$migrationClass = $this->migrationRegister->getMigration( $migrationId );
+			$migration      = give( $migrationClass );
+			$migration->run();
+			// Save migration status
+			$migrationLog->setStatus( MigrationLogStatus::SUCCESS );
+			$migrationLog->setError( null );
```

Comment:
Calling a setter without a parameter doesn't feel right. Assuming `setError( null )` is clearing errors, what about `clearError()` with no parameter?

Reply:
@kjohnson yeah, that doesn't look right, but I'm not sure if `clearError()` is really needed here. `clearError` implies that error already happened and we want to clear the error, but that's not true.

Reply:
@alaca in that case, what is the intention of `setError( null )`?

Reply:
@ravinderk no, we can't remove that because if migration already ran, and it failed, we should update the message as well by calling `setError(null)` or `clearError`.

Reply:
Am I understanding correctly that an error may or may not have already been set, so passing null is `maybe clearing an error`? We can clarify that intention with `maybeClearError()` or something like `resetErrors()` - anyway, the name is an implementation detail, the important thing is that we clearly communicate our intention.

Reply:
@kjohnson to explicitly set error for that migration to `null` :smiley:  
The truth is, we don't care about migration status at that point, which can be `success` or `failed`. But, if migration already ran before, and it failed, we have to update the error message. That's why using `setError` there makes sense.

Reply:
In that case, `setError( null )` does make sense 😆 

Might be worth a short comment to clarify, since it totally makes sense if you know that is the intention.

Reply:
I usually try to refactor code to clarify with naming conventions where possible, but in some cases you just need a quick comment for the next dev (which may be a future you!).

Reply:
@kjohnson sure, I'll do that.  Thanks.